Sources/BoostBLEKit/HubProperty.swift

```diff
@@ -8,56 +8,120 @@
+    
+    public enum Encoding {
+        case bool
+        case uint8
+        case int8
+        case string
+        case versionNumber
+        case lwpVersionNumber
+        case macAddress
+    }
     
     public enum Value {
-        
+        case bool(Bool)
+        case uint8(UInt8)
+        case int8(Int8)
         case string(String)
-        case integer(Int)
-        
-        public var stringValue: String {
-            switch self {
-            case .string(let string):
-                return string
-            case .integer(let integer):
-                return "\(integer)"
-            }
-        }
-        
-        public var intValue: Int? {
-            switch self {
-            case .string:
-                return nil
-            case .integer(let integer):
-                return integer
-            }
-        }
     }
     
-    public func value(from data: Data) -> Value? {
+    public var encoding: Encoding {
         switch self {
         case .advertisingName:
+            return .string
+        case .button:
+            return .bool
+        case .firmwareVersion:
+            return .versionNumber
+        case .hardwareVersion:
+            return .versionNumber
+        case .rssi:
+            return .int8
+        case .batteryVoltage:
+            return .uint8
+        case .batteryType:
+            return .uint8
+        case .manufacturerName:
+            return .string
+        case .radioFirmwareVersion:
+            return .string
+        case .legoWirelessProtocolVersion:
+            return .lwpVersionNumber
+        case .systemTypeId:
+            return .uint8
+        case .HWNetworkId:
+            return .uint8
+        case .primaryMacAddress:
+            return .macAddress
+        case .secondaryMacAddress:
+            return .macAddress
+        case .hardwareNetworkFamily:
+            return .uint8
+        }
+    }
+    
+    public func value(from data: Data) -> Value? {
+        switch encoding {
+        case .bool:
+            return .bool(data[0] == 0 ? false : true)
+            
+        case .uint8:
+            return .uint8(data[0])
+            
+        case .int8:
+            return .int8(data.withUnsafeBytes({ $0.load(as: Int8.self) }))
+            
+        case .string:
             guard let string = String(data: data, encoding: .utf8) else { return nil }
             return .string(string)
             
-        case .firmwareVersion:
-            guard data.count == 4 else { return nil }
-            var version = data.reversed().map { String(format: "%02x", $0) }.joined(separator: "")
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 1))
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 3))
-            version.insert(".", at: version.index(version.startIndex, offsetBy: 6))
-            return .string(version)
+        case .versionNumber:
+            precondition(data.count == 4)
```

Comment:
Could you use `guard data.count == 4 else { return nil }` instead? I'd not like to crash if the hub responds any invalid data.